docs/extend/forms.md

```diff
@@ -0,0 +1,114 @@
+# Forms and Requests
+
+In this article, we'll go over some frontend tools that are available to us for building and managing forms, as well how to send HTTP requests via Flarum.
+
+## Form Components
+
+As with any interactive site, you will likely want to include forms in some pages and modals.
+Flarum provides some components to make building (and styling!) these forms easier.
+Please see the linked API documentation for each of these to learn more about its accepted attrs.
+
+- The [`flarum/components/FieldSet` component](https://api.docs.flarum.org/js/master/class/src/common/components/fieldset.js~fieldset) wraps its children in a HTML fieldset tag, with a legend.
+- The [`flarum/components/Select` component](https://api.docs.flarum.org/js/master/class/src/common/components/select.js~select) is a stylized select input.
+- The [`flarum/components/Switch`](https://api.docs.flarum.org/js/master/class/src/common/components/switch.js~switch) and [`flarum/components/Checkbox` components](https://api.docs.flarum.org/js/master/class/src/common/components/checkbox.js~checkbox) are stylized checkbox input components. Their `loading` attr can be set to `true` to show a loading indicator.
+- The [`flarum/components/Button` component](https://api.docs.flarum.org/js/master/class/src/common/components/button.js~button) is a stylized button, and is used frequently throughout Flarum.
+
+You'll typically want to assign logic for reacting to input changes via attrs, not external listeners. For example:
```

Comment:
I'm a bit confused by that sentence. I thought it referred to the way data must be transferred up a component child, but that's not what the example demonstrate.

Should it just say that you can register event listeners via Mithril's `on*` attributes?